src/jshint.js

```diff
@@ -3373,15 +3373,21 @@ var JSHINT = (function() {
         element_after_rest = true;
       }
       while (!checkPunctuators(state.tokens.next, ["]"])) {
-        advance(",");
-        if (checkPunctuators(state.tokens.next, ["]"])) {
-          // Trailing comma
-          break;
+        if (checkPunctuators(state.tokens.next, ["="])) {
+          advance("=");
+          expression(10);
```

Comment:
Do we want to do any destructuring on the default expression? I feel like if we get an undefined literal we should emit a warning, just because it's redundant